gcs-fetcher/pkg/fetcher/fetcher.go

```diff
@@ -651,23 +653,19 @@ func (gf *Fetcher) fetchFromManifest(ctx context.Context) (err error) {
 	return nil
 }
 
-func (gf *Fetcher) copyFileFromZip(file *zip.File) (err error) {
-	sourceReader, err := file.Open()
-	if err != nil {
-		return fmt.Errorf("failed to open source file %q: %v", file.Name, err)
-	}
+func (gf *Fetcher) copyFile(name string, mode os.FileMode, rc io.ReadCloser) (err error) {
 	defer func() {
-		if cerr := sourceReader.Close(); cerr != nil {
-			err = fmt.Errorf("Failed to close file %q: %v", file.Name, cerr)
+		if cerr := rc.Close(); cerr != nil {
+			err = fmt.Errorf("Failed to close file %q: %v", name, cerr)
 		}
 	}()
 
-	targetFile := filepath.Join(gf.DestDir, file.Name)
-	if err := gf.ensureFolders(targetFile); err != nil {
-		return fmt.Errorf("failed to create folders for file %q: %v", targetFile, err)
+	targetFile := filepath.Join(gf.DestDir, name)
+	if err := gf.OS.MkdirAll(filepath.Base(targetFile), mode); err != nil {
```

Comment:
Can you just use gf.DestDir here (instead of filepath.Base())?

Reply:
I don't think so. `gf.DestDir` is something like `/workspace`, while `name` might include a path to the file, e.g., `path/to/file.go`, so you need to `mkdir `/workspace/path/to/` before writing `file.go`. Does that match your understanding of this code?